src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -3843,7 +3843,7 @@ private BoundExpression BindNullCoalescingOperator(BinaryExpressionSyntax node,
 
             if ((object)optRightType != null && optRightType.IsDynamic())
             {
-                var leftPlaceholder = new BoundValuePlaceholder(leftOperand.Syntax, optLeftType).MakeCompilerGenerated();
+                var leftPlaceholder = new BoundValuePlaceholder(leftOperand.Syntax, optLeftType0).MakeCompilerGenerated();
```

Comment:
>optLeftType0 [](http://example.com/codeflow?start=84&length=12)

This change looks suspicious. The type doesn't match the conversion that we are classifying below.

Reply:
You're right. I missed that we're still using `leftOperand` in `ClassifyConversionFromExpression`, so they don't match :-/